portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,49 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> commentParts;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    String json = convertToJsonUsingGson(commentParts);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getParameter(request, "name", "Anonymous");
+    String email = getParameter(request, "email", "n/a");
+    String comment = getParameter(request, "comment-input", "");
+
+    // Add input into an ArrayList.
+    commentParts = new ArrayList<>();
+    commentParts.add(name);
+    commentParts.add(email);
+    commentParts.add(comment);
+
+    // Redirect back to the blog page.
+    response.sendRedirect("/blog.html");
+  }
+
+  /**
+   * Converts a List of strings into a JSON string using the Gson library.
+   */
+  private String convertToJsonUsingGson(List<String> commentParts) {
+    String json = new Gson().toJson(commentParts);
+    return json;
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
```

Comment:
im aware this is in the starter/sample code, but just as fyi
https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment